### PR TITLE
Indicate what entity(s) are blocking the use of the paired endpoint suffix

### LINKED ISSUE #6339: Entity(s) causing path suffix conflicts are not indicated
## Feature description

In a couple cases recently users have reported the inability to select the Legacy Reader or Path Suffix paired URL structures and getting this notice:

> There is a post, term, user, or some other entity that is already using the “amp” URL slug. For this reason, you cannot currently use the path suffix or legacy reader paired URL structures.

* https://wordpress.org/support/topic/my-amp-urls-return-a-blank-screen-when-i-click-on-google-serps-mobile/#post-14498701
* https://wordpress.org/support/topic/500-xxx-error-paired-url-structure-featured-images-not-showing/

We don't expose the conflict in the UI at present.

For example, if I add this plugin code:

```php
add_action('init', function () {
	register_taxonomy( 'amp', 'post' );
});
```

The settings screen shows:

image

The conflict is not shown, but the conflicts _are_ returned in the API response.

In the console doing:

```js
(await wp.apiFetch({path:'/amp/v1/options'})).endpoint_path_slug_conflicts
```

Results in:

image

So we just need to amend the notice with a list of what the conflicts are. This can be done in an unordered list.

The REST API endpoint should return the taxonomy slug, taxonomy translated name, and term edit link for each entity if available. When an option says “unavailable due to slug conflict” there should be a clear reference to the above notice.

https://github.com/ampproject/amp-wp/blob/3bb15f3d660c4401b958147430664d07eb1e03e2/src/PairedRouting.php#L337-L390

Also, I just realized that we should also check if the `amp_get_slug()` exists among `WP_Rewrite::$endpoints` and indicate that as another conflict.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by westonruter**
## Summary

Fixes #6339

Before | After
-------|------
image | image

### Testing

To test the functionality:

* Create a post with the slug `amp`.
* Create a page with the slug `amp`.
* Create a post tag with the slug `amp`.
* Create a category with the slug `amp`.
* Create a user with the username of `amp`.

Then:

* Register a post type with `amp` as the rewrite slug.
* Register a taxonomy with `amp` as the rewrite slug.
* Register an `amp` rewrite endpoint.

These can be achieved with this plugin code:

```php
add_action(
	'init',
	function () {
		add_rewrite_endpoint( 'amp', E_ALL );
		register_post_type( 'amp-post',
			[
				'labels' => [
					'name'          => 'AMP Posts',
					'singular_name' => 'AMP Post',
				],
				'rewrite' => [
					'slug' => 'amp'
				],
			]
		);
		register_taxonomy( 'amp-tax', 'post', [
			'labels' => [
				'name'          => 'AMP Taxes',
				'singular_name' => 'AMP Tax',
			],
			'rewrite' => [
				'slug' => 'amp'
			],
		] );
	}
);
```

## Checklist

- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [ ] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: src/PairedRouting.php [hunk 1]
```diff
@@ -372,7 +408,13 @@ public function get_endpoint_path_slug_conflicts() {
 				||
 				isset( $post_type->rewrite['slug'] ) && $post_type->rewrite['slug'] === $amp_slug
```
Comment (pierlon) [inline src/PairedRouting.php]:
```suggestion
				( isset( $post_type->rewrite['slug'] ) && $post_type->rewrite['slug'] === $amp_slug )
```

File: src/PairedRouting.php [hunk 2]
```diff
@@ -382,9 +424,43 @@ public function get_endpoint_path_slug_conflicts() {
 				||
 				isset( $taxonomy->rewrite['slug'] ) && $taxonomy->rewrite['slug'] === $amp_slug
```
Comment (pierlon) [inline src/PairedRouting.php]:
```suggestion
				( isset( $taxonomy->rewrite['slug'] ) && $taxonomy->rewrite['slug'] === $amp_slug )
```

File: assets/src/settings-page/paired-url-structure.js [hunk 3]
```diff
@@ -88,6 +88,173 @@ function getCustomPairedStructureSources( sources ) {
...
+									sprintf( __( '(ID: %d)', 'amp' ), entity.id )
+								}
+							</small>
+						</li>
+					) )
+				) }
+
+				{
+					conflicts.user && (
+						[ conflicts.user ].map( ( entity ) => (
```
Comment (pierlon) [inline assets/src/settings-page/paired-url-structure.js]: Any reason for wrapping this in an array and then `map`ping it?

File: assets/src/settings-page/paired-url-structure.js [hunk 4]
```diff
@@ -88,6 +88,173 @@ function getCustomPairedStructureSources( sources ) {
...
+								{ ': ' + entity.name }
+								{ ' ' }
+								<small>
+									{
+										/* translators: %d is entity ID */
+										sprintf( __( '(ID: %d)', 'amp' ), entity.id )
+									}
+								</small>
+							</li>
+						) )
```
Comment (pierlon) [inline assets/src/settings-page/paired-url-structure.js]:
```suggestion
					<li>
							{
								conflicts.user.edit_link ? (
									<a href={ conflicts.user.edit_link } target="_blank" rel="noreferrer">
										{ __( 'User', 'amp' ) }
									</a>
								) : (
									__( 'User', 'amp' )
								)
							}
							{ ': ' + conflicts.user.name }
							{ ' ' }
							<small>
								{
									/* translators: %d is entity ID */
									sprintf( __( '(ID: %d)', 'amp' ), conflicts.user.id )
								}
							</small>
						</li>
```

File: assets/src/settings-page/paired-url-structure.js [hunk 5]
```diff
@@ -88,6 +88,173 @@ function getCustomPairedStructureSources( sources ) {
...
+					)
+				}
+
+				{
+					conflicts.post_type && (
+						<li key="post_type">
```
Comment (pierlon) [inline assets/src/settings-page/paired-url-structure.js]:
```suggestion
						<li>
```

File: assets/src/settings-page/paired-url-structure.js [hunk 6]
```diff
@@ -88,6 +88,173 @@ function getCustomPairedStructureSources( sources ) {
...
+									sprintf( __( '(name: %s)', 'amp' ), conflicts.post_type.name )
+								}
+							</small>
+						</li>
+					)
+				}
+
+				{
+					conflicts.taxonomy && (
+						<li key="taxonomy">
```
Comment (pierlon) [inline assets/src/settings-page/paired-url-structure.js]:
```suggestion
						<li>
```

File: assets/src/settings-page/paired-url-structure.js [hunk 7]
```diff
@@ -88,6 +88,173 @@ function getCustomPairedStructureSources( sources ) {
...
+									/* translators: %s is entity name */
+									sprintf( __( '(name: %s)', 'amp' ), conflicts.taxonomy.name )
+								}
+							</small>
+						</li>
+					)
+				}
+
+				{ conflicts.rewrite && (
+					<li key="rewrite">
```
Comment (pierlon) [inline assets/src/settings-page/paired-url-structure.js]:
```suggestion
					<li>
```

File: assets/src/settings-page/paired-url-structure.js [hunk 8]
```diff
@@ -88,6 +88,173 @@ function getCustomPairedStructureSources( sources ) {
...
+						}
+					</li>
+				) }
+			</ul>
+		</AMPNotice>
+	);
+}
+SlugConflictsNotice.propTypes = {
+	slug: PropTypes.string.isRequired,
+	conflicts: PropTypes.object.isRequired,
```
Comment (pierlon) [inline assets/src/settings-page/paired-url-structure.js]: Could specify the shape of the object here:

```suggestion
	conflicts: PropTypes.shape( {
		post_type: PropTypes.object,
		posts: PropTypes.arrayOf( PropTypes.object ),
		rewrite: PropTypes.arrayOf( PropTypes.string ),
		taxonomy: PropTypes.object,
		terms: PropTypes.arrayOf( PropTypes.object ),
		user: PropTypes.object,
	} ).isRequired,
```

**Review by pierlon — CHANGES_REQUESTED** [review]

File: assets/src/settings-page/paired-url-structure.js (hunk 3 above)
Comment (pierlon) [reply assets/src/settings-page/paired-url-structure.js]: (See suggestion below)

Comment (westonruter) [reply assets/src/settings-page/paired-url-structure.js]: Mainly just to make the style similar to the preceding two, but it's not needed.

**Review by pierlon — APPROVED** [review]

**Pull request merged**